test/create_env_cluster_topics_test.go

```diff
@@ -76,30 +57,201 @@ func TestEnvClusterTopic(t *testing.T) {
+	})
+}
 
-		output = terraform.Output(t, runOptions, "kafka_topic_name")
-		a.Equal(output, "squad_raw_service_example_1_entity")
+// Logic to Validate Total ACL Provision via Confluent CLI
+func validateTotalKafkaACLCountWithCLI(t *testing.T) int {
+	output := runCLICommand(t, "confluent", "kafka", "acl", "list", "-o", "json")
+	log.Printf("Kafka ACL CLI Output: %s", output)
 
-		output = terraform.Output(t, runOptions, "kafka_topic_name")
-		a.Equal(output, "squad_raw_service_example_1_entity")
+	var acls []map[string]interface{}
+	err := json.Unmarshal([]byte(output), &acls)
+	assert.NoError(t, err, "Failed to parse Kafka ACL JSON output")
 
-		output = terraform.Output(t, runOptions, "bigquery_connector_id")
-		a.NotEmpty(output)
+	aclCount := len(acls)
+	log.Printf("Actual ACL count: %d", aclCount)
+	return aclCount
+}
 
-		output = terraform.Output(t, runOptions, "connector_gcs_sink_connector_id")
-		a.NotEmpty(output)
-	})
+// Logic to validate individual ACL combinations (ResourceType and Operation) via Confluent CLI
+func validateIndividualKafkaACLDetails(t *testing.T, expectedACLs []struct {
+	ResourceType string
+	Operation    string
+	Count        int
+}) {
+
+	output := runCLICommand(t, "confluent", "kafka", "acl", "list", "-o", "json")
+	var acls []map[string]interface{}
+	err := json.Unmarshal([]byte(output), &acls)
+	assert.NoError(t, err, "Failed to parse Kafka ACL JSON output")
+
+	// Count ACLs by ResourceType and Operation
+	actualCounts := make(map[string]int)
+	for _, acl := range acls {
+		key := fmt.Sprintf("%s|%s", acl["resource_type"], acl["operation"])
+		actualCounts[key]++
+	}
+
+	// Validate actual counts against expected values
+	for _, expected := range expectedACLs {
+		key := fmt.Sprintf("%s|%s", expected.ResourceType, expected.Operation)
+		actualCount := actualCounts[key]
+		assert.Equal(t, expected.Count, actualCount, fmt.Sprintf(
+			"Expected %d ACLs for ResourceType: %s, Operation: %s, but found %d",
+			expected.Count, expected.ResourceType, expected.Operation, actualCount,
+		))
+	}
+
+	log.Println("Individual ACL combination validation successful.")
+}
+
+// Logic to validate Terraform outputs with Confluent CLI
+func validateTerraformOutputWithCLI(t *testing.T, envID, environmentName, kafkaClusterType, kafkaClusterID, kafkaClusterBasicName, bigQueryConnectorID, gcsSinkConnectorID, kafkaTopicName1, kafkaTopicName2, topicServiceAccountID string) {
+	a := assert.New(t)
+
+	// Validate Environment ID and Name
+	log.Println("Validating Environment ID and Name...")
+	envOutput := runCLICommand(t, "confluent", "env", "describe", envID, "-o", "json")
+	var envDetails map[string]interface{}
+	a.NoError(json.Unmarshal([]byte(envOutput), &envDetails), "Failed to parse environment details")
+	a.Equal(envID, envDetails["id"], "Environment ID mismatch")
+	a.Equal(environmentName, envDetails["name"], "Environment Name mismatch")
```

Comment:
we're still validating Terraform outputs against the Confluent CLI. We need to be matching **inputs** to the resulting infrastructure.